benchmarks/data_utils.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+from matplotlib import pyplot as plt
+import os
+
+import torch
+from torch.utils.data import Dataset
+
+import torchvision
+import torchvision.transforms.functional as TF
+
+
+
+
+###
+# The dataset initializes with the file path to frame pairs
+# and loads the image when get image is called. Did this back when planning to take frames 
+# from entire video. Now the data is stored as as only 2 frames, so not really a necessary step
+###
+class FramePairDataset(Dataset):
+    def __init__(self, path_to_folder, class_types=['control', 'mdivi', 'llo'], transform=None):
+        self.targets= []
+        self.vid_path = []
+        self.transform = transform
+        # self.transform_input = transform_input
+        self.class_types = class_types
+
+        print(path_to_folder)
+        
+
+        name_constraint = lambda x: 'normalized' in x
+        for label in self.class_types:
+            target =  self.class_types.index(label) #class 0,1,2
+            path = os.path.join(path_to_folder, label)
+            files = os.listdir(path)
+            for file_name in files:
+                    if  name_constraint(file_name):
+                        self.vid_path.append(os.path.join(path, file_name))
+                        self.targets.append(target)
+        
+    def __len__(self):
+        return len(self.targets)
+
+    
+    def __getitem__(self, idx):
+        target_class = self.targets[idx]
+        vid = np.load(self.vid_path[idx])
+        first_frame = vid[0]
+        last_frame = vid[-1] 
```

Comment:
That should be pulling from `/data/ornet/single_cell_cnns`, which should be the static dataset you want. Just wrote it that way so it could work on complete videos if need be.